### PR TITLE
search contexts: show highlight bubble

### LINKED ISSUE #19839: Draw attention to the context selector when initially shown
## Problem

In our initial implementation for the contexts selector and user-added repositories, we defaulted to the user's personal context and showed that context once the user added repos. This was effective at drawing the user's attention to the contexts selector once it became visible (your own name appearing in the UI tends to be effective at drawing attention). However, the personal context as default caused some conflicts with existing code examples, so we changed to using the global context by default instead. As a consequence, when the context selector is show in the search input after the user adds repositories, it's less visible and may not be connected to the action they just took.

## Proposal

Since the search contexts selector's content is self-explanatory, our goal is to draw attention to the selector as an element. I propose we re-use the current "tour" UI to highlight it the first time the contexts selector is shown.

It would look something like this (re-used current design assets for the tour—may look a bit different in production): 

CleanShot 2021-04-08 at 20 18 06@2x

This would also be a good opportunity to add a docs page for search contexts.

### PR DESCRIPTION
**Pull request by novoselrok**
Show search contexts highlight bubble when the dropdown is initially shown.

Fixes #19839

### CODE REVIEW

**Comment by quinnkeast** [comment]
Looking good!

I think we should make it so that clicking on the contexts selector dismisses the highlight, as well.

**Comment by novoselrok** [comment]
I'll update the TODO link, once https://github.com/sourcegraph/sourcegraph/pull/19916 is merged.

File: client/web/src/search/input/SearchContextDropdown.tsx [hunk 1]
```diff
@@ -113,6 +124,24 @@ export const SearchContextDropdown: React.FunctionComponent<SearchContextDropdow
...
+                            Learn more
+                        </a>
+                    </div>
+                    <div className="d-flex justify-content-end">
+                        <button type="button" className="btn btn-sm" onClick={closeHighlightBubble}>
+                            Close
+                        </button>
+                    </div>
+                </div>
+            )}
```
Comment (limitedmage) [inline client/web/src/search/input/SearchContextDropdown.tsx:144]: Can we move this into a generic component that can be reused? We need to build the same UI for Code Monitoring.

File: client/web/src/search/input/SearchContextDropdown.tsx [hunk 2]
```diff
@@ -113,6 +124,24 @@ export const SearchContextDropdown: React.FunctionComponent<SearchContextDropdow
                     />
                 </DropdownMenu>
             </Dropdown>
+            {showSearchContextHighlightBubble && !hasSeenHighlightBubble && (
+                <div className="search-context-dropdown__highlight-bubble">
```
Comment (limitedmage) [inline client/web/src/search/input/SearchContextDropdown.tsx:128]: Can this be a Reach tooltip/modal instead so we don't have issues with stacking contexts? Reach uses React Portals to simplify the DOM.

Comment (limitedmage) [reply client/web/src/search/input/SearchContextDropdown.tsx:128]: Alternatively we can use Shepherd which we are already using for the search onboarding tour

**Comment by novoselrok** [comment]
@limitedmage I decided to use the Shepherd tour to implement the bubble, thanks for the suggestion. It uses the same styling as the search onboarding tour. Once we want to implement a similar single-step tour somewhere else, we can "componentize" this solution.

// cc @quinnkeast

**Comment by quinnkeast** [comment]
Looks great! One last thought: maybe "Learn more" should open in a new tab, to preserve the user's place on Sourcegraph?

**Review by lguychard — APPROVED** [review]

**Review by quinnkeast — APPROVED** [review]

**Review by limitedmage — COMMENTED** [review]
Post-merge review looks great!

**Pull request merged**